src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -250,20 +251,11 @@ export const txHistoryApi = createApi({
 
             const state = getState() as State
             const txState = state.txHistory.txs
-            /**
-             * TODO(0xdef1cafe): perf improvement - change this check back to use
-             * a flatMap of txs indexed on the txState?.byAccountIdAssetId?.[accountId] object
-             *
-             * checking against the flat list of all tx ids is slower, but a stop gap.
-             *
-             * we currently can't index txs against asset ids we don't know about
-             * e.g. erc721 and erc1155 assets
-             *
-             * after we rewrite unchained client tx parsers for these, we can index txs correctly,
-             * and go back to using those shorter lists per account
-             */
-            // the existing tx indexes for this account
-            const existingTxIndexes = txState?.ids
+
+            const existingTxIndexes = Object.values(
```

Comment:
this should be safe to revert back to flat map now that we are processing nft txs/transfers. correct me if I am wrong

Reply:
Yeah I reckon web `txState` should be able to handle all kind of Txs unchained does now

Reply:
beautiful ty!

plz ensure no regressions here w.r.t. overfetching tx history - should only fetch 1 page for each account after tx history is cached for a wallet